include/deal.II/lac/solver.h

```diff
@@ -21,7 +21,10 @@
 #include <deal.II/lac/vector_memory.h>
 #include <deal.II/lac/solver_control.h>
 
+// Ignore deprecation warnings for auto_ptr.
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/signals2.hpp>
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
```

Comment:
The short answer is 'no'.

That is a very good question that took some GCC research to answer. I originally used the bundled boost (the force bundled boost option) but I also tested with my system boost (1.58).

As far as I can tell, if one uses a boost headers in a standard location (mine is in `/usr/include/`, so paths not specified with `-I`) then GCC automatically supresses warnings. My copy of deal.II which was configured to use my system boost did not raise any such deprecation warnings. When I symbolically linked my vendor-supplied headers into my user copy of deal.II (i.e., installed under `/home/drwells/Applications/dealii-dev/`) the warnings re-appeared.